autograder-core/src/test/java/de/firemage/autograder/core/CheckTest.java

```diff
@@ -91,7 +94,7 @@ Stream<DynamicTest> createCheckTest() throws URISyntaxException, IOException {
             folders = paths.toList();
         }
 
-        TempLocation tempLocation = TempLocation.random();
+        TempLocation tempLocation = new TempLocationImpl();
```

Comment:
I liked the static method more

Reply:
The constructors are cleaner for reflection, so additional static methods felt redundant. But we can add them back in if you prefer.

Reply:
I would like to have the factory method back.